### PR TITLE
add package.json to MagicOnion.Client.Unity / fix asmdef missing MessagePack.Annotations

### PR DESCRIPTION
**Pull request by guitarrapc**
## TL;DR

* feature: Added package.json to MagicOnion.Client.Unity.
* fix: MagicOnion.Client.asmdef and MagicOnion.Abstract.asmdef to include `MessagePack.Annotations` references for latest Message Package Package.

## Summary

* Now MagicOnion Unity package supports UPM, it work on local file ref.
* Latest MessagePack split asmdef for MessagePack.Annotations and it cause reference errors, this is fixed with PR.
* Changed MagicOnion.Client.Unity's Project Structure for `Assets/Scripts`. Now MagicOnion related packages are under `Assets/Scripts/MagicOnion` directory.
image

* Update MessagePack reference in MagicOnion.Client.Unity to latest MessagePack.Unity.2.2.85.unitypackage, this fix MessagePack.Annotations reference on Unity 2018.4.12f1 and below. (no error on 2018.4.13f1 and above)

Unity 2020.2.0a
image

### CODE REVIEW

**Review by mayuki — APPROVED** [review]
👍

File: src/MagicOnion.Client.Unity/Assets/Scripts/MagicOnion/package.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+  {
+    "name": "com.cysharp.magiconion",
+    "displayName": "MagicOnion",
+    "version": "4.0.4",
+    "unity": "2018.3",
+    "description": "Unified Realtime/API framework for .NET platform and Unity.",
+    "keywords": [
```
Comment (guitarrapc) [inline src/MagicOnion.Client.Unity/Assets/Scripts/MagicOnion/package.json:7]: is keyword enought?
